packages/shared/src/modules/activity-logs/libs/types/activity-log-get-all-item-analytics-response-dto.type.ts

```diff
@@ -1,6 +1,7 @@
 type ActivityLogGetAllItemAnalyticsResponseDto = {
 	commitsNumber: (number | string)[];
 	contributorId: string;
+	contributorIsHidden: boolean;
 	contributorName: string;
```

Comment:
so what is with the question regarding moving this props inside contributor object

Reply:
I haven't found any issues with but I still can't figure out why I'm getting the error `[Cannot destructure property 'id' of 'gitEmail.contributor' as it is null.]`. I'm trying to understand why by using the following query:

```
const activityLogs = await this.activityLogModel
			.query()
			.withGraphFetched("[gitEmail.contributor, project, createdByUser]")
			.modifyGraph("gitEmail.contributor", (builder) => {
				builder.select("id", "name").where("isHidden", false);
			})
			.whereBetween("activity_logs.date", [startDate, endDate])
			.orderBy("date");
```

I see that contributor is coming back as `null`.

Reply:
Oh, I also faced similar problem. I guess the problem is how graph fetcing works, and how you used it. 
You are trying to filter main query (`SELECT ... FROM activity_logs`) by the condition, that contributor is hidden, but your code just selects all the activity logs in some date range, and then fetches to those selected activity logs contibutors data  (other stuff doesn't matter right now), but you add additional `.where` filter, which just replaced contributors with `isHidden=true` to null in activity_logs where it is the case.
Try to use [withGraphJoined](https://vincit.github.io/objection.js/api/query-builder/eager-methods.html#withgraphjoined) instead of fetched, and then add `.where` filter to main query

Reply:
Something like
```ts
const activityLogs = await this.activityLogModel
	.query()
	.withGraphJoined("[gitEmail.contributor, project, createdByUser]")
	.modifyGraph("gitEmail.contributor", (builder) => {
		builder.select("id", "name");
	})
	.where("contributor.isHidden", false)
	.whereBetween("activity_logs.date", [startDate, endDate])
	.orderBy("date");
```

Reply:
![image](https://github.com/user-attachments/assets/198d10d0-be78-459c-98ae-4036d11a9d3c)

Reply:
Using this info probably the query should looks like
```ts
const activityLogs = await this.activityLogModel
	.query()
	.withGraphFetched("[project, createdByUser]")
	.withGraphJoined("gitEmail.contributor")
	.where("gitEmail:contributor.isHidden", false)
	.whereBetween("activity_logs.date", [startDate, endDate])
	.orderBy("date");
```

and also we need to add something which will do
```ts
.modifyGraph("gitEmail.contributor", (builder) => {
	builder.select("id", "name");
})
```
cause this method won't work with joining the graph. Or will be, I'm not sure.

Reply:
Thanks a lot, I've tried smth similar to this, last time it didn't give the needed result. But hope this time it will :)